lib/std/compilesettings.nim

```diff
@@ -0,0 +1,22 @@
+#
+#
+#           The Nim Compiler
+#        (c) Copyright 2020 Nim Contributors
+#
+#    See the file "copying.txt", included in this
+#    distribution, for details about the copyright.
+#
+
+proc compileSetting*(option: string): string {.
+  compiletime, noSideEffect.} = discard
+## Can be used to get a string compile-time option. Example:
+##
+## .. code-block:: Nim
+##   const nimcache = compileSetting("nimcachedir")
+
+proc compileSettingSeq*(option: string): seq[string] {.
+  compiletime, noSideEffect.} = discard
+## Can be used to get a multi-string compile-time option. Example:
```

Comment:
Indentation is off and it needs to document what "option" can be.

Reply:
instead of documenting (not DRY, not helpful if user wants to iterate over all values), how about doing what I suggested here: https://github.com/nim-lang/Nim/pull/13010#issuecomment-593141491
```nim
type CompilerSettings = object
  arguments: seq[string]
  outFile: string
  outDir: string
```
(see note above for details)